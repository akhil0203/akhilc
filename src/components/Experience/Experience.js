import React from 'react';
import { Container, Col, Row } from 'reactstrap';

import pagerdutyImg from '../../images/thumbnails/companies/pagerduty.png';
import wattpadImg from '../../images/thumbnails/companies/wattpad.png';
import capitalOneImg from '../../images/thumbnails/companies/capital-one.png';
import senecaImg from '../../images/thumbnails/companies/seneca.png';
import bmoImg from '../../images/thumbnails/companies/bmo.png';
import vincentAssociatesImg from '../../images/thumbnails/companies/vincent-associates.png';

import './Experience.scss';

const Experience = () => {
  return (
    <Container className="experience" fluid>
      <Col>
        <Row>
          <h2>Experience</h2>
        </Row>
        <Col>
          <Row>
            <Col sm={3}>
              <Row>
                <a href="https://bit.ly/pagerduty_home" target="pagerduty">
                  <img src={pagerdutyImg} alt="pagerduty" />
                </a>
              </Row>
              <Row>
                <h3>Software Engineer Intern</h3>
              </Row>
              <Row>
                <h4>PagerDuty, Toronto ON</h4>
              </Row>
              <Row>
                <p>September 2019 - December 2019</p>
              </Row>
            </Col>
            <Col sm={3}>
              <Row>
                <a href="https://bit.ly/wattpad_home" target="wattpad">
                  <img src={wattpadImg} alt="wattpad" />
                </a>
              </Row>
              <Row>
                <h3>Software Engineer Intern</h3>
              </Row>
              <Row>
                <h4>Wattpad, Toronto ON</h4>
              </Row>
              <Row>
                <p>January 2019 - April 2019</p>
              </Row>
            </Col>
            <Col sm={3}>
              <Row>
                <a href="https://bit.ly/capitalone_home" target="capital-one">
                  <img src={capitalOneImg} alt="capital-one" />
                </a>
              </Row>
              <Row>
                <h3>Software Engineer Intern</h3>
              </Row>
              <Row>
                <h4>Capital One, Toronto ON</h4>
              </Row>
              <Row>
                <p>May 2018 - August 2018</p>
              </Row>
            </Col>
          </Row>

          <Row>
            <Col sm={3}>
              <Row>
                <a href="https://bit.ly/seneca_home" target="seneca">
                  <img src={senecaImg} alt="seneca" />
                </a>
              </Row>
              <Row>
                <h3>Tutor</h3>
              </Row>
              <Row>
                <h4>Seneca, Toronto ON</h4>
              </Row>
              <Row>
                <p>September 2017 - April 2018</p>
              </Row>
            </Col>
            <Col sm={3}>
              <Row>
                <a href="https://bit.ly/bmo_home" target="bmo">
                  <img src={bmoImg} alt="bmo" />
                </a>
              </Row>
              <Row>
                <h3>Software Engineer Intern</h3>
              </Row>
              <Row>
                <h4>Bank of Montreal, Toronto ON</h4>
              </Row>
              <Row>
                <p>May 2017 - September 2017</p>
              </Row>
            </Col>
            <Col sm={3}>
              <Row>
                <a
                  href="https://bit.ly/vincentinc_home"
                  target="vincent-associates"
                >
                  <img src={vincentAssociatesImg} alt="vincent-associates" />
                </a>
              </Row>
              <Row>
                <h3>Software Engineer</h3>
              </Row>
              <Row>
                <h4>Vincent Associates, Toronto ON</h4>
              </Row>
              <Row>
                <p>May 2015 - August 2016</p>
              </Row>
            </Col>
          </Row>
        </Col>
      </Col>
    </Container>
  );
};

export default Experience;
