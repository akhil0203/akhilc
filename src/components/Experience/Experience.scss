.experience {
  background-image: url(../../images/backgrounds/experience.jpg);
  background-size: cover;
  display: flex;

  > .col {
    padding: 150px 0 150px 0;
  }

  .col > .row:not(:last-child) {
    margin-bottom: 40px;
  }

  .row {
    justify-content: center;
    align-items: center;

    h2 {
      color: #fff;
      font-size: 40px;
      font-weight: 700;
      text-transform: uppercase;
    }

    h3 {
      color: #fff;
      font-size: 30px;
      font-weight: 700;
      text-align: center;
    }

    h4 {
      color: #fff;
      font-size: 16px;
      font-weight: 700;
      text-align: center;
    }

    p {
      color: #fff;
      font-size: 16px;
      font-weight: 700;
      text-align: center;
    }
  }
}
