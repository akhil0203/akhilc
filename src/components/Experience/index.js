import React from 'react';
import { Container, Col, Row } from 'reactstrap';
import { ParallaxBanner } from 'react-scroll-parallax';
import { Event } from '../Shared/Tracking';

import backgroundImg from '../../images/backgrounds/experience.jpg';

import pagerdutyImg from '../../images/thumbnails/companies/pagerduty.png';
import wattpadImg from '../../images/thumbnails/companies/wattpad.png';
import capitalOneImg from '../../images/thumbnails/companies/capital-one.png';
import senecaImg from '../../images/thumbnails/companies/seneca.png';
import bmoImg from '../../images/thumbnails/companies/bmo.png';
import vincentAssociatesImg from '../../images/thumbnails/companies/vincent-associates.png';

import './styles.scss';

const Experience = () => {
  return (
    <ParallaxBanner
      className="experience-parallax"
      layers={[
        {
          image: JSON.stringify(backgroundImg),
          amount: 0.25,
        },
      ]}
    >
      <Container className="experience" fluid>
        <Col>
          <Row>
            <h2>Experience</h2>
          </Row>
          <Col>
            <Row>
              <Col sm={3}>
                <Row>
                  <a
                    href="https://bit.ly/pagerduty_home"
                    target="pagerduty"
                    onClick={() =>
                      Event(
                        'Experience',
                        'Clicked PagerDuty company link',
                        'PagerDuty company link',
                      )
                    }
                  >
                    <img src={pagerdutyImg} alt="pagerduty" />
                  </a>
                </Row>
                <Row>
                  <h3>Software Engineer Intern</h3>
                </Row>
                <Row>
                  <h4>PagerDuty, Toronto ON</h4>
                </Row>
                <Row>
                  <p>September 2019 - December 2019</p>
                </Row>
              </Col>
              <Col sm={3}>
                <Row>
                  <a
                    href="https://bit.ly/wattpad_home"
                    target="wattpad"
                    onClick={() =>
                      Event(
                        'Experience',
                        'Clicked Wattpad company link',
                        'Wattpad company link',
                      )
                    }
                  >
                    <img src={wattpadImg} alt="wattpad" />
                  </a>
                </Row>
                <Row>
                  <h3>Software Engineer Intern</h3>
                </Row>
                <Row>
                  <h4>Wattpad, Toronto ON</h4>
                </Row>
                <Row>
                  <p>January 2019 - April 2019</p>
                </Row>
              </Col>
              <Col sm={3}>
                <Row>
                  <a
                    href="https://bit.ly/capitalone_home"
                    target="capital-one"
                    onClick={() =>
                      Event(
                        'Experience',
                        'Clicked Capital One company link',
                        'Capital One company link',
                      )
                    }
                  >
                    <img src={capitalOneImg} alt="capital-one" />
                  </a>
                </Row>
                <Row>
                  <h3>Software Engineer Intern</h3>
                </Row>
                <Row>
                  <h4>Capital One, Toronto ON</h4>
                </Row>
                <Row>
                  <p>May 2018 - August 2018</p>
                </Row>
              </Col>
            </Row>

            <Row>
              <Col sm={3}>
                <Row>
                  <a
                    href="https://bit.ly/seneca_home"
                    target="seneca"
                    onClick={() =>
                      Event(
                        'Experience',
                        'Clicked Seneca company link',
                        'Seneca company link',
                      )
                    }
                  >
                    <img src={senecaImg} alt="seneca" />
                  </a>
                </Row>
                <Row>
                  <h3>Tutor</h3>
                </Row>
                <Row>
                  <h4>Seneca, Toronto ON</h4>
                </Row>
                <Row>
                  <p>September 2017 - April 2018</p>
                </Row>
              </Col>
              <Col sm={3}>
                <Row>
                  <a
                    href="https://bit.ly/bmo_home"
                    target="bmo"
                    onClick={() =>
                      Event(
                        'Experience',
                        'Clicked BMO company link',
                        'BMO company link',
                      )
                    }
                  >
                    <img src={bmoImg} alt="bmo" />
                  </a>
                </Row>
                <Row>
                  <h3>Software Engineer Intern</h3>
                </Row>
                <Row>
                  <h4>Bank of Montreal, Toronto ON</h4>
                </Row>
                <Row>
                  <p>May 2017 - September 2017</p>
                </Row>
              </Col>
              <Col sm={3}>
                <Row>
                  <a
                    href="https://bit.ly/vincentinc_home"
                    target="vincent-associates"
                    onClick={() =>
                      Event(
                        'Experience',
                        'Clicked Vincent Associates company link',
                        'Vincent Associates company link',
                      )
                    }
                  >
                    <img src={vincentAssociatesImg} alt="vincent-associates" />
                  </a>
                </Row>
                <Row>
                  <h3>Software Engineer</h3>
                </Row>
                <Row>
                  <h4>Vincent Associates, Toronto ON</h4>
                </Row>
                <Row>
                  <p>May 2015 - August 2016</p>
                </Row>
              </Col>
            </Row>
          </Col>
        </Col>
      </Container>
    </ParallaxBanner>
  );
};

export default Experience;
