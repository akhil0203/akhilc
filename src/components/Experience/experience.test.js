import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Experience from '.';

it('matches the snapshot', () => {
  const tree = renderer.create(<Experience />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders the Experience component without crashing', () => {
  shallow(<Experience />);
});

it('renders the main header', () => {
  const wrapper = shallow(<Experience />);

  expect(wrapper.find('h2').html()).toEqual('<h2>Experience</h2>');
});

it('renders details for TunnelBear', () => {
  const wrapper = shallow(<Experience />);

  expect(
    wrapper
      .find('Col>Row>Col')
      .at(0)
      .html(),
  ).toEqual(
    '<div class="col-sm-3"><div class="row"><a href="https://bit.ly/tunnelbear_home" target="tunnelbear"><img src="tunnelbear.png" alt="tunnelbear"/></a></div><div class="row"><h3>Site Reliability Engineer</h3></div><div class="row"><h4>TunnelBear, Toronto ON</h4></div></div>',
  );
});

it('renders details for PagerDuty', () => {
  const wrapper = shallow(<Experience />);

  expect(
    wrapper
      .find('Col>Row>Col')
      .at(1)
      .html(),
  ).toEqual(
    '<div class="col-sm-3"><div class="row"><a href="https://bit.ly/pagerduty_home" target="pagerduty"><img src="pagerduty.png" alt="pagerduty"/></a></div><div class="row"><h3>Software Engineer Intern</h3></div><div class="row"><h4>PagerDuty, Toronto ON</h4></div></div>',
  );
});

it('renders details for Wattpad', () => {
  const wrapper = shallow(<Experience />);

  expect(
    wrapper
      .find('Col>Row>Col')
      .at(2)
      .html(),
  ).toEqual(
    '<div class="col-sm-3"><div class="row"><a href="https://bit.ly/wattpad_home" target="wattpad"><img src="wattpad.png" alt="wattpad"/></a></div><div class="row"><h3>Software Engineer Intern</h3></div><div class="row"><h4>Wattpad, Toronto ON</h4></div></div>',
  );
});

it('renders details for Capital One', () => {
  const wrapper = shallow(<Experience />);

  expect(
    wrapper
      .find('Col>Row>Col')
      .at(3)
      .html(),
  ).toEqual(
    '<div class="col-sm-3"><div class="row"><a href="https://bit.ly/capitalone_home" target="capital-one"><img src="capital-one.png" alt="capital-one"/></a></div><div class="row"><h3>Software Engineer Intern</h3></div><div class="row"><h4>Capital One, Toronto ON</h4></div></div>',
  );
});

it('renders details for BMO', () => {
  const wrapper = shallow(<Experience />);

  expect(
    wrapper
      .find('Col>Row>Col')
      .at(4)
      .html(),
  ).toEqual(
    '<div class="col-sm-3"><div class="row"><a href="https://bit.ly/bmo_home" target="bmo"><img src="bmo.png" alt="bmo"/></a></div><div class="row"><h3>Software Engineer Intern</h3></div><div class="row"><h4>Bank of Montreal, Toronto ON</h4></div></div>',
  );
});

it('renders details for Vincent Associates', () => {
  const wrapper = shallow(<Experience />);

  expect(
    wrapper
      .find('Col>Row>Col')
      .at(5)
      .html(),
  ).toEqual(
    '<div class="col-sm-3"><div class="row"><a href="https://bit.ly/vincentinc_home" target="vincent-associates"><img src="vincent-associates.png" alt="vincent-associates"/></a></div><div class="row"><h3>Software Engineer</h3></div><div class="row"><h4>Vincent Associates, Toronto ON</h4></div></div>',
  );
});
