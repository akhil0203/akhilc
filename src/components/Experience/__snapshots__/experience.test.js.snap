// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div
  className="experience container-fluid"
>
  <div
    className="col"
  >
    <div
      className="row"
    >
      <h2>
        Experience
      </h2>
    </div>
    <div
      className="col"
    >
      <div
        className="row"
      >
        <div
          className="col-sm-3"
        >
          <div
            className="row"
          >
            <a
              href="https://bit.ly/tunnelbear_home"
              onClick={[Function]}
              target="tunnelbear"
            >
              <img
                alt="tunnelbear"
                src="tunnelbear.png"
              />
            </a>
          </div>
          <div
            className="row"
          >
            <h3>
              Software Engineer
            </h3>
          </div>
          <div
            className="row"
          >
            <h4>
              TunnelBear, Toronto ON
            </h4>
          </div>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-sm-3"
        >
          <div
            className="row"
          >
            <a
              href="https://bit.ly/pagerduty_home"
              onClick={[Function]}
              target="pagerduty"
            >
              <img
                alt="pagerduty"
                src="pagerduty.png"
              />
            </a>
          </div>
          <div
            className="row"
          >
            <h3>
              Software Engineer Intern
            </h3>
          </div>
          <div
            className="row"
          >
            <h4>
              PagerDuty, Toronto ON
            </h4>
          </div>
        </div>
        <div
          className="col-sm-3"
        >
          <div
            className="row"
          >
            <a
              href="https://bit.ly/wattpad_home"
              onClick={[Function]}
              target="wattpad"
            >
              <img
                alt="wattpad"
                src="wattpad.png"
              />
            </a>
          </div>
          <div
            className="row"
          >
            <h3>
              Software Engineer Intern
            </h3>
          </div>
          <div
            className="row"
          >
            <h4>
              Wattpad, Toronto ON
            </h4>
          </div>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-sm-3"
        >
          <div
            className="row"
          >
            <a
              href="https://bit.ly/capitalone_home"
              onClick={[Function]}
              target="capital-one"
            >
              <img
                alt="capital-one"
                src="capital-one.png"
              />
            </a>
          </div>
          <div
            className="row"
          >
            <h3>
              Software Engineer Intern
            </h3>
          </div>
          <div
            className="row"
          >
            <h4>
              Capital One, Toronto ON
            </h4>
          </div>
        </div>
        <div
          className="col-sm-3"
        >
          <div
            className="row"
          >
            <a
              href="https://bit.ly/bmo_home"
              onClick={[Function]}
              target="bmo"
            >
              <img
                alt="bmo"
                src="bmo.png"
              />
            </a>
          </div>
          <div
            className="row"
          >
            <h3>
              Software Engineer Intern
            </h3>
          </div>
          <div
            className="row"
          >
            <h4>
              Bank of Montreal, Toronto ON
            </h4>
          </div>
        </div>
        <div
          className="col-sm-3"
        >
          <div
            className="row"
          >
            <a
              href="https://bit.ly/vincentinc_home"
              onClick={[Function]}
              target="vincent-associates"
            >
              <img
                alt="vincent-associates"
                src="vincent-associates.png"
              />
            </a>
          </div>
          <div
            className="row"
          >
            <h3>
              Software Engineer
            </h3>
          </div>
          <div
            className="row"
          >
            <h4>
              Vincent Associates, Toronto ON
            </h4>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
